scarlet2/initialization.py

```diff
@@ -0,0 +1,331 @@
+# ---------------------------------------------------------------------------- #
+# Initialization routine for Scarlet2. We define function allowing for the     #
+# initialization of source morphologies and spectrums of indentified sources.  #
+#                                                                              #
+# For morphologies, we input the observational image with the locations of the #
+# sources calculated. Then we fit a 2D Gaussian to the cutout and optimize for #
+# the best fit with a maximum amplitute of 1.                                  #
+#                                                                              #
+# For spectrums, we input the observational image as with the morphology, we   #
+# just take the pixel value of the center of the source for each band.         #
+# ---------------------------------------------------------------------------- #
+
+import numpy as np
+import numpy.ma as ma
+import jax.numpy as jnp
+import jax
+from .plot import cut_square_box
+from .morphology import GaussianMorphology
+from jaxopt import ScipyBoundedMinimize
+import dm_pix
+from functools import partial
+
+
+# function to calculate values of perimeter pixels
+def perimeter_values(array_2d):
+    """Return the values of the perimeter pixels of a 2D array."""
+    rows, cols = len(array_2d), len(array_2d[0])
+    perimeter_values = []
+    # Top row
+    perimeter_values.extend(array_2d[0])
+    # Bottom row
+    if rows > 1:
+        perimeter_values.extend(array_2d[-1])
+    # Left and right columns, excluding corners to avoid double counting
+    if cols > 1:
+        perimeter_values.extend(array_2d[i][0] for i in range(1, rows - 1))
+        if (
+            cols > 2
+        ):  # Check if there are more than 2 columns to avoid duplicate corner values
+            perimeter_values.extend(array_2d[i][-1] for i in range(1, rows - 1))
+
+    return perimeter_values
+
+
+# create a 2D Gaussian for simple feature free Initializations
+def gaussian_2d(x, y, mu_x, mu_y, sigma_x, sigma_y):
+    """2D Gaussian function.
+    Parameters
+    ----------
+    x : jnp.ndarray
+    y : jnp.ndarray
+    mu_x : float mean of the Gaussian in the x direction
+    mu_y : float mean of the Gaussian in the y direction
+    sigma_x : float standard deviation of the Gaussian in the x direction
+    sigma_y : float standard deviation of the Gaussian in the y direction
+
+    Returns
+    -------
+    jnp.ndarray 2D Gaussian profile
+    """
+    return jnp.exp(
+        -((x - mu_x) ** 2 / (2 * sigma_x**2) + (y - mu_y) ** 2 / (2 * sigma_y**2))
+    )
```

Comment:
We have that implemented in `GaussianMorphology`, so you should use that instead.